File: app/src/main/java/com/example/gitbusca/service/GitHubService.java
package com.example.gitbusca.service;

import com.example.gitbusca.model.GitCatalogo;
import com.example.gitbusca.model.Pull;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GitHubService {

    public static final String BASE_URL = "https://api.github.com/";
    public static final String REPOS = "repos/";





    //@GET("repositories?q=language:Java&sort=stars&page=1")
    @GET("search/repositories")
    Call<GitCatalogo> listarCatalogo(
            @Query("q") String query,
            @Query("sort") String sort,
            @Query("page") int page);

    @GET("pulls")
    Call<List<Pull>> listarPulls();

}

File: app/src/main/java/com/bassul/mel/app/endpoint/GithubService.kt
package com.bassul.mel.app.endpoint

import com.bassul.mel.app.feature.pullRequestsList.repository.model.PullRequestListResponse
import com.bassul.mel.app.feature.repositoriesList.repository.model.RepositoriesListResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubService {

    companion object {
        val BASE_URL = "https://api.github.com/"
    }

    @GET("search/repositories?q=language:java&sort=stargazers_count")
    fun fetchRepository(@Query("page") id: Int): Call<RepositoriesListResponse>

    @GET("repos/{login}/{repositoryName}/pulls")
    fun fetchPullRequest(
        @Path("login") login: String,
        @Path("repositoryName") repositoryName: String
    ): Call<List<PullRequestListResponse>>
}